contracts/glv/glvDeposit/GlvDepositUtils.sol

```diff
@@ -193,14 +198,48 @@ library GlvDepositUtils {
             );
         }
 
-        cache.receivedMarketTokens = _processMarketDeposit(params, glvDeposit);
-        cache.mintAmount = _getMintAmount(params.dataStore, params.oracle, glvDeposit, cache.receivedMarketTokens);
+        // glvTokenPrice should be calculated before glv receives GM tokens
+        uint256 glvTokenPrice = GlvUtils.getGlvTokenPrice(params.dataStore, params.oracle, glvDeposit.glv(), true);
+        cache.receivedMarketTokens = _processMarketDeposit(params, glvDeposit, params.glvVault);
```

Comment:
should there be a validation for min tokens for first glv deposit? see minMarketTokensForFirstDepositKey and certora audit for details

mainly it is to prevent a user from distorting the initial GLV price, for market tokens the risk is that the price could be distorted to a very large value, making a newly created market unusable

for GLV, i'm not sure if there is a similar risk or other implications